SDBlockDevice.cpp

```diff
@@ -587,6 +587,16 @@ bd_size_t SDBlockDevice::get_program_size() const
     return _block_size;
 }
 
+bd_size_t SDBlockDevice::get_erase_size() const
+{
+    return _erase_size;
```

Comment:
Hold on, I think this should actually be "_block_size". The "_erase_size" in the SD driver doesn't match the "erase_size" of the block device API. The SD driver's "_erase_size" is actually used by the "trim" function.

@deepikabhavnani, isn't that right?

Actually does SD need an overload of `get_erase_size` since it doesn't overload either name?

Reply:
Erase size is SD driver is actual erase size of SD device which may not be equal to block size.
Erase functionality in SD driver is provided by trim.  

SD driver does not provide overload for erase, since it is not required. But in order to have successful trim (erase operation) size should be in multiple of erase else internally SD driver will perform erase of more than a block and user might assume its corrupted SD card.

We can provide erase size in form of get_trim_size()  / get_erase_size().. 
@geky - What will be preferred?

Reply:
@geky - You are right, get_erase_size should be _block_size for filesystem. Fat file system gets the sector size from erase size and not program/read size.
https://github.com/ARMmbed/mbed-os/blob/02af950f62ecfbd0013c81f9e0a644da508088d4/features/filesystem/fat/FATFileSystem.cpp#L174

If erase size is actually needed, we can add it as get_trim_size